packages/devtools_app/lib/src/http/http_request_data.dart

```diff
@@ -553,18 +117,65 @@ class DartIOHttpRequestData extends HttpRequestData {
     };
   }
 
+  @override
+  String? get contentType {
+    if (responseHeaders == null || responseHeaders![_contentTypeKey] == null) {
+      return null;
+    }
+    return responseHeaders![_contentTypeKey].toString();
+  }
+
+  @override
+  String get type {
+    var mime = contentType;
+    if (mime == null) return 'http';
+
+    // Extract the MIME from `contentType`.
+    // Example: "[text/html; charset-UTF-8]" --> "text/html"
+    mime = mime.split(';').first;
+    if (mime.startsWith('[')) {
+      mime = mime.substring(1);
+    }
+    if (mime.endsWith(']')) {
+      mime = mime.substring(0, mime.length - 1);
+    }
+    return _extensionFromMime(mime);
+  }
+
+  /// Extracts the extension from [mime], with overrides for shortened
+  /// extenstions of common types (e.g., jpe -> jpeg).
+  String _extensionFromMime(String mime) {
+    final extension = extensionFromMime(mime);
+    if (extension == 'jpe') {
+      return 'jpeg';
+    }
+    if (extension == 'htm') {
+      return 'html';
+    }
+    // text/plain -> conf
+    if (extension == 'conf') {
+      return 'txt';
+    }
+    return extension;
+  }
+
   @override
   String get method => _request.method;
 
+  @override
+  int? get port {
+    final Map<String, dynamic>? connectionInfo = general[_connectionInfoKey];
+    return connectionInfo != null ? connectionInfo[_localPortKey] : null;
+  }
+
   /// True if the HTTP request hasn't completed yet, determined by the lack of
   /// an end event.
   @override
   bool get inProgress =>
       _hasError ? !_request.isRequestComplete : !_request.isResponseComplete;
 
   /// All instant events logged to the timeline for this HTTP request.
```

Comment:
This comment still references the timeline, but you've removed that in this PR, no?